client/components/TestRun/index.jsx

```diff
@@ -447,8 +431,23 @@ const TestRun = () => {
             !rendererState ||
             !testResult.scenarioResults ||
             rendererState.commands.length !== testResult.scenarioResults.length
-        )
-            return testResult;
+        ) {
+            // Mapping unexpected behaviors to expected TestRenderer downstream format
```

Comment:
REVIEWERS: I had to do this so that the TestRenderer would receive consistent data.